src/app/pages/buyer/buyer-bid-list/buyer-bid-list.component.ts

```diff
@@ -1,23 +1,81 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { AppService } from 'app/app.service';
-import { Bid } from 'app/app.model';
+import { Bid, Buyer } from 'app/app.model';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'wm-buyer-bid-list',
   templateUrl: './buyer-bid-list.component.html',
   styleUrls: ['./buyer-bid-list.component.scss']
 })
-export class BuyerBidListComponent implements OnInit {
+export class BuyerBidListComponent implements OnInit, OnDestroy {
   public bids: Bid[] = [];
+  private routeParamSubscr: Subscription;
+  public buyer: Buyer;
 
-  constructor(public appServ: AppService) {}
+  constructor(public appServ: AppService, private route: ActivatedRoute, private router: Router) {}
 
   ngOnInit() {
-    this.appServ.getBidsForBuyer(this.appServ.loggedInUserInfo['id']).subscribe(data => {
+    this.listenToRouteParams();
+  }
+
+  /**
+   * @description listen to the route params for changes in buyer id
+   */
+  private listenToRouteParams() {
```

Comment:
I think we need not listen to route params as we dont have internal transitions i.e., transitions from same route to same route with params changed. and so, none of the below written new methods are required. 

When ever the route is changed from buyer bid list route to admin buyer list route, this component gets killed. and the next time, `ngOnInit` hook would get triggered anyway. 

This component should not have too many changes is what i feel. We should not optimize it until its needed. Thoughts?

Reply:
I agree with you. I guess I have written it too futuristic-ally. I understand your point. My initial concern was when a buyer changes the buyer id in the route and refreshes the page(they would have access to other buyers' bid lists) considering we use the same component for buyer and admin. Starting with that I put resolve to the other cases as well. Considering this, I think we should retain just the initial case that I mentioned and leave out listening to the params, what say?

Reply:
Yeah. i dont think listening to params is needed. 

> they would have access to other buyers' bid lists

is this being handled? if yes, how?

Reply:
I will remove the param listening part. On page load, buyer id in the param is validated against the persona and only authorized users are able to view the buyer's bids, else are redirected to their own bids(the logged in user's buyer id)

Reply:
cool. can you do the same validation for seller too?

Reply:
Can add the same for the seller too. However there would arise a constraint that post this navigation, we would need to listen to the params to provide the necessary data which we thought otherwise not to listen previously.

Reply:
i agree.